src/components/hint/index.js

```diff
@@ -0,0 +1,335 @@
+// TODO: Add support to custom hint rendering
+// TODO: Add animation
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import {View as AnimatableView} from 'react-native-animatable';
+import {StyleSheet} from 'react-native';
+import {BaseComponent} from '../../commons';
+import View from '../view';
+import Text from '../text';
+import Image from '../image';
+import {Typography, Spacings, Colors, BorderRadiuses, AnimatableManager} from '../../style';
+import {Constants} from '../../helpers';
+
+const sideTip = require('./assets/hintTipSide.png');
+const middleTip = require('./assets/hintTipMiddle.png');
+
+const DEFAULT_COLOR = Colors.blue30;
+const DEFAULT_HINT_DISTANCE = Spacings.s4;
+const DEFAULT_EDGE_SPACE = Spacings.s5;
+const HINT_POSITIONS = {
+  TOP: 'top',
+  BOTTOM: 'bottom'
+};
+
+AnimatableManager.loadAnimationDefinitions({
+  hintAppearDown: {
+    from: {opacity: 0, translateY: 20},
+    to: {opacity: 1, translateY: 0}
+  },
+  hintAppearUp: {
+    from: {opacity: 0, translateY: -20},
+    to: {opacity: 1, translateY: 0}
+  }
+});
+
+class Hint extends BaseComponent {
+  static displayName = 'Hint';
+
+  static propTypes = {
+    /**
+     * Control the visibility of the hint
+     */
+    visible: PropTypes.bool,
+    /**
+     * The hint background color
+     */
+    color: PropTypes.string,
+    /**
+     * The hint message
+     */
+    message: PropTypes.string,
+    /**
+     * The hint message custom style
+     */
+    messageStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * Icon to show next to the hint's message
+     */
+    icon: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
+    /**
+     * The icon's style
+     */
+    iconStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * The hint's position
+     */
+    position: PropTypes.oneOf(_.values(HINT_POSITIONS)),
+    /**
+     * Show side tips instead of the middle tip
+     */
+    useSideTip: PropTypes.bool,
+    /**
+     * The hint's border radius
+     */
+    borderRadius: PropTypes.number,
+    /**
+     * Hint space from screen edges
+     */
+    edgeSpace: PropTypes.number,
```

Comment:
you used 'distance' in the next prop, why use 'space' here?

Reply:
cause this is the horizontal space from the edges, we can change it maybe to margins. 
The other prop has more explicit meaning, it's the distance of the hint from the target, it's not just a space

Reply:
and 'hintDistance' is vertical distance/space/margin from the target view. I don't see much difference, I just think it is unclear when you name props that are quite similar in synonyms...

Reply:
Yes it's vertical, What would you call these props?